AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,692 @@
+                    let tap = UITapGestureRecognizer(target: self, action: #selector(ViewController.recognizeLocalAudio))
+                    preview.addGestureRecognizer(tap)
+                    view.addSubview(preview);
+                }
+            }
+        } else if (localAudioTrack != nil) {
+            logMessage(messageText: "Front camera is not available, using microphone only.")
+        } else {
+            logMessage(messageText: "Failed to create audio track!")
+        }
+    }
+
+    func setupRemoteVideoView(publication: TVIRemoteVideoTrackPublication) {
+        // Create a `TVIVideoView` programmatically, and add to our `UIStackView`
+        if let remoteView = TVIVideoView.init(frame: CGRect.zero, delegate:nil) {
+            // We will bet that a hash collision between two unique SIDs is very rare.
+            remoteView.tag = publication.trackSid.hashValue
+
+            // `TVIVideoView` supports scaleToFill, scaleAspectFill and scaleAspectFit
+            // scaleAspectFit is the default mode when you create `TVIVideoView` programmatically.
+            remoteView.contentMode = .scaleAspectFit;
+
+            // Double tap to change the content mode.
+            let recognizerDoubleTap = UITapGestureRecognizer(target: self, action: #selector(ViewController.changeRemoteVideoAspect))
+            recognizerDoubleTap.numberOfTapsRequired = 2
+            remoteView.addGestureRecognizer(recognizerDoubleTap)
+
+            // Single tap to recognize remote audio.
+            let recognizerTap = UITapGestureRecognizer(target: self, action: #selector(ViewController.recognizeRemoteAudio))
+            recognizerTap.require(toFail: recognizerDoubleTap)
+            remoteView.addGestureRecognizer(recognizerTap)
+
+            // Start rendering, and add to our stack.
+            publication.remoteTrack?.addRenderer(remoteView)
+            self.remoteViewStack.addArrangedSubview(remoteView)
+        }
+    }
+
+    func removeRemoteVideoView(publication: TVIRemoteVideoTrackPublication) {
+        let viewTag = publication.trackSid.hashValue
+        if let remoteView = self.remoteViewStack.viewWithTag(viewTag) {
+            // Stop rendering, we don't want to receive any more frames.
+            publication.remoteTrack?.removeRenderer(remoteView as! TVIVideoRenderer)
+            // Automatically removes us from the UIStackView's arranged subviews.
+            remoteView.removeFromSuperview()
+        }
+    }
+
+    func changeRemoteVideoAspect(gestureRecognizer: UIGestureRecognizer) {
+        guard let remoteView = gestureRecognizer.view else {
+            print("Couldn't find a view attached to the tap recognizer. \(gestureRecognizer)")
+            return;
+        }
+
+        if (remoteView.contentMode == .scaleAspectFit) {
+            remoteView.contentMode = .scaleAspectFill
+        } else {
+            remoteView.contentMode = .scaleAspectFit
+        }
```

Comment:
no `scaleToFill` ?

Reply:
I guess I'm not a fan of stretching video to make people look very fat or thin.